crates/astria-bridge-withdrawer/src/withdrawer/ethereum/watcher.rs

```diff
@@ -208,6 +210,16 @@ fn address_from_string(s: &str) -> Result<ethers::types::Address> {
     Ok(address.into())
 }
 
+fn event_to_action(event_with_metadata: EventWithMetadata) -> eyre::Result<Action> {
+    let action = BridgeUnlockAction {
+        to: event_with_metadata.event.sender.to_fixed_bytes().into(),
+        amount: event_with_metadata.event.amount.as_u128(),
+        memo: event_with_metadata.event.memo.to_vec(), // TODO: store block number and tx hash here
+        fee_asset_id: Id::from_denom(FEE_ASSET_ID),
```

Comment:
the `event.sender` is an evm address - we need to parse the `event.memo` to get the destination sequencer address (or check if it's an ics20withdrawal) 

also thinking amount needs a "precision" as well, as going to the evm for example we multiply by some precision so here we'd need to divide by that precision

memo should also be updated to store the evm block number and tx hash

Reply:
i propose the evm tx memo looks like:
```
{
    "astria": {
         "recipient_address": "astriaaddress",
    },
    "forward": {
         "receiver": "destinationchainbech32address",
         "port": "transfer",
         "channel": "channel-123"
    }
} 
```

where `forward` is set only if this is an ics20 withdrawal, and `astria` can be unset. if it's a withdrawal to a sequencer account, then `astria` is set and `forward` is unset

Reply:
i also propose the memo for `BridgeUnlockAction` looks like
```
{
     "rollup_block_number: 99,
     "rollup_tx_hash": "tx_hash"
}
```

tx_hash comes from `event_with_metadata`.

Reply:
no more evm tx memo, see https://github.com/astriaorg/astria/pull/1101

for ibc withdrawals, channel/port are hardcoded. the ibc packet memo should contain whatever is in the `Ics20Withdrawal` event memo plus additional data:
```
{
     "rollup_block_number": 99,
     "rollup_tx_hash": "tx_hash",
     "rollup_sender_address": 0x...
}
```